pkg/tsdb/tempo/trace_transform.go

```diff
@@ -205,75 +168,67 @@ func getAttributeVal(attr pdata.AttributeValue) interface{} {
+func getSpanTags(span pdata.Span, instrumentationLibrary pdata.InstrumentationLibrary) []*KeyValue {
 	var tags []*KeyValue
+
+	libraryTags := getTagsFromInstrumentationLibrary(instrumentationLibrary)
+	if libraryTags != nil {
+		tags = append(tags, libraryTags...)
+	}
 	span.Attributes().Range(func(key string, attr pdata.AttributeValue) bool {
 		tags = append(tags, &KeyValue{Key: key, Value: getAttributeVal(attr)})
 		return true
 	})
+
+	status := span.Status()
+	possibleNilTags := []*KeyValue{
+		getTagFromSpanKind(span.Kind()),
+		getTagFromStatusCode(status.Code()),
+		getErrorTagFromStatusCode(status.Code()),
+		getTagFromStatusMsg(status.Message()),
+		getTagFromTraceState(span.TraceState()),
+	}
+
+	for _, tag := range possibleNilTags {
+		if tag != nil {
+			tags = append(tags, tag)
+		}
+	}
 	return tags
 }
 
-func getSpanKind(spanKind pdata.SpanKind) string {
-	var tagStr string
-	switch spanKind {
-	case pdata.SpanKindClient:
-		tagStr = string(tracetranslator.OpenTracingSpanKindClient)
-	case pdata.SpanKindServer:
-		tagStr = string(tracetranslator.OpenTracingSpanKindServer)
-	case pdata.SpanKindProducer:
-		tagStr = string(tracetranslator.OpenTracingSpanKindProducer)
-	case pdata.SpanKindConsumer:
-		tagStr = string(tracetranslator.OpenTracingSpanKindConsumer)
-	case pdata.SpanKindInternal:
-		tagStr = string(tracetranslator.OpenTracingSpanKindInternal)
-	default:
-		return ""
-	}
+func getTagsFromInstrumentationLibrary(il pdata.InstrumentationLibrary) []*KeyValue {
+	var keyValues []*KeyValue
+	return keyValues
+}
 
-	return tagStr
+func getTagFromSpanKind(spanKind pdata.SpanKind) *KeyValue {
+	return nil
 }
 
-func getTraceState(traceState pdata.TraceState) string {
-	if traceState != pdata.TraceStateEmpty {
-		return string(traceState)
-	}
-	return ""
+func getTagFromStatusCode(statusCode pdata.StatusCode) *KeyValue {
+	return nil
 }
 
-func spanEventsToLogs(events pdata.SpanEventSlice) []*TraceLog {
-	if events.Len() == 0 {
-		return nil
-	}
+func getErrorTagFromStatusCode(statusCode pdata.StatusCode) *KeyValue {
+	return nil
+}
 
-	logs := make([]*TraceLog, 0, events.Len())
-	for i := 0; i < events.Len(); i++ {
-		event := events.At(i)
-		fields := make([]*KeyValue, 0, event.Attributes().Len()+1)
-		if event.Name() != "" {
-			fields = append(fields, &KeyValue{
-				Key:   tracetranslator.TagMessage,
-				Value: event.Name(),
-			})
-		}
-		event.Attributes().Range(func(key string, attr pdata.AttributeValue) bool {
-			fields = append(fields, &KeyValue{Key: key, Value: getAttributeVal(attr)})
-			return true
-		})
-		logs = append(logs, &TraceLog{
-			Timestamp: float64(event.Timestamp()) / 1_000_000,
-			Fields:    fields,
-		})
-	}
+func getTagFromStatusMsg(statusMsg string) *KeyValue {
+	return nil
+}
+
+func getTagFromTraceState(traceState pdata.TraceState) *KeyValue {
+	return nil
+}
 
-	return logs
+func spanEventsToLogs(events pdata.SpanEventSlice) []*TraceLog {
+	return nil
```

Comment:
Is this what you mean by commented code? Was this method emptied to make the build pass due to the otel library downgrade?

Reply:
yeah, i think we'll need to do something like this to reset those files and see if they build:

```
git checkout main pkg/tsdb/tempo pkg/infra/httpclient/httpclientprovider/tracing_middleware.go
```